yogstation/code/modules/mining/lavaland/bloodbook.dm

```diff
@@ -0,0 +1,875 @@
+/obj/item/bloodbook/proc/splosion(mob/living/user, mob/living/target)
+	var/magnitude = ((target.maxHealth/150) + 1)
+	for(var/turf/closed/mineral/mine in range(target, magnitude))
+		mine.attempt_drill()
+	for(var/mob/living/collateral in view(target, magnitude))
+		var/explosiondam = target.maxHealth/10
+		if((collateral == user) || (collateral == target))
+			continue
+		if(!isanimal(collateral) && explosiondam >= 20)
+			explosiondam = 20
+		var/result = (severitycalc(collateral, explosiondam+1))
+		addtimer(CALLBACK(src, PROC_REF(inflict), user, collateral, explosiondam, FALSE))
+		if(result)
+			addtimer(CALLBACK(src, PROC_REF(splosion), user, collateral))
+
+/obj/item/bloodbook/examine(datum/source, mob/user, list/examine_list)
+	. = ..()
+	. += span_notice("It has [blocks] charges available.")
+
+/obj/item/bloodbook/proc/crystallize(mob/living/target)
+	if(isanimal(target))
+		target.color = "#ff0000"
+		target.alpha = 100
+
+/obj/item/bloodbook/proc/shatter(mob/living/simple_animal/target)
+	var/ore = pick(orelist)
+	if(!isanimal(target))
+		if(target.mind)
+			return
+	if(istype(target, /mob/living/simple_animal/hostile/retaliate/goat/king))
+		return //cant trap people in the room forever
+	if((!istype(target, /mob/living/simple_animal/hostile/asteroid/hivelordbrood/legion)))
+		for(var/i =1 to 3)
+			new ore(target.loc)
+	if(istype(target, /mob/living/simple_animal/hostile/megafauna/legion))
+		target.adjustBruteLoss(target.health)
+		return
+	playsound(target, "shatter", 70, 1)
+	if(ismegafauna(target))
+		var/mob/living/simple_animal/hostile/megafauna/L = target
+		for(var/V in L.guaranteed_butcher_results)
+			new V(target.loc)
+		L.guaranteed_butcher_results = null
+		for(var/V in L.butcher_results)
+			new V(target.loc)
+		L.butcher_results = null
+		if(target.stat == DEAD)
+			target.gib()
+			return //no double dipping
+		target.dust(TRUE, FALSE, TRUE)
+		return
+	if(isanimal(target))
+		target.drop_loot()
+		target.loot = null
+	target.gib()
+
+//animation stuff
+
+
+/obj/item/bloodbook/proc/animatepick(mob/living/user, mob/living/target, var/severity)
+	if(isanimal(target))
+		var/mob/living/simple_animal/L = target
+		L.toggle_ai(AI_OFF)
+	if(severity == 1)
+		if(ismegafauna(target))
+			if(istype(target, /mob/living/simple_animal/hostile/megafauna/legion))
+				headmassage(user,target)
+				return
+			if(istype(target, /mob/living/simple_animal/hostile/megafauna/blood_drunk_miner))
+				if(prob(75))
+					workphobia(user,target)
+					return
+			pick(mash(user, target), spiritbomb(user, target))
+			return
+		pick(tantrum(user, target), falling(user, target), damnedfang(user, target), dunk(user, target), shrink(user, target), redshot(user, target), vortex(user, target))
+		return
+	if(severity > 1)
+		pick(inducingemission(user,target), concavehead(user,target))
+		return
+
+//kill animations
+
+/obj/item/bloodbook/proc/tantrum(mob/living/user, mob/living/target, phase = 1, obj/hand, list/slamming)
+	switch(phase)
+		if(1)
```

Comment:
These float numbers should be defines that you will undefine after this proc, same for the ones under.